src/App.tsx

```diff
@@ -15,48 +16,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+  RESET = '',
+}
+
+interface FilterParams {
+  sortValue: SortType;
+  isReverse: boolean;
+}
+
+function sortGoods(
+  goods: string[],
+  { sortValue, isReverse }: FilterParams,
+): string[] {
+  const preparedGoods = goods;
+
+  preparedGoods.sort((goodA, goodB) => {
+    switch (sortValue) {
+      case SortType.ALPHABET:
+        return goodA.localeCompare(goodB);
+
+      case SortType.LENGTH:
+        return goodA.length - goodB.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReverse, setIsReverse] = useState(false);
+  const [sortValue, setSortValue] = useState(SortType.RESET);
+  const sortedGoods = sortGoods([...goodsFromServer], { sortValue, isReverse });
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
+          onClick={() => setSortValue(SortType.ALPHABET)}
           type="button"
-          className="button is-info is-light"
+          className={cn('button is-info',
+            { 'is-light': sortValue !== SortType.ALPHABET })}
         >
           Sort alphabetically
         </button>
 
         <button
+          onClick={() => setSortValue(SortType.LENGTH)}
           type="button"
-          className="button is-success is-light"
+          className={cn('button is-success',
+            { 'is-light': sortValue !== SortType.LENGTH })}
         >
           Sort by length
         </button>
 
         <button
+          onClick={() => setIsReverse(!isReverse)}
           type="button"
-          className="button is-warning is-light"
+          className={cn('button is-warning',
+            { 'is-light': !isReverse })}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(sortValue || isReverse) && (
```

Comment:
You can create a variable for (sortValue || isReverse) and put it there
Try not to mix logical operators as this unnecessarily complicates your code